app/tests/spec/lib/marketing-email-client.js

```diff
@@ -0,0 +1,130 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'chai',
+  'sinon',
+  'lib/marketing-email-client',
+  'lib/promise',
+  'lib/constants',
+  'lib/xhr'
+],
+function (chai, sinon, MarketingEmailClient, p, Constants, Xhr) {
+  'use strict';
+
+  var assert = chai.assert;
+
+  describe('lib/marketing-email-client', function () {
+    var BASE_URL = 'https://basket.mozilla.com';
+
+    var client;
+    var xhrMock;
+
+    beforeEach(function () {
+      // Xhr has no constructor
+      xhrMock = Object.create(Xhr);
+      xhrMock.ajax = function () {
+        return p();
+      };
+
+      client = new MarketingEmailClient({
+        xhr: xhrMock,
+        baseUrl: BASE_URL
+      });
+    });
+
+    describe('getPreferencesUrl', function () {
+      it('returns a preferences URL for the user', function () {
+        sinon.stub(xhrMock, 'ajax', function () {
+          return p({
+            token: 'users_uuid'
+          });
+        });
+
+        return client.getPreferencesUrl('token')
+          .then(function (url) {
+            assert.equal(url, BASE_URL + '/user/users_uuid');
+
+            var request = xhrMock.ajax.args[0][0];
+            assert.equal(request.url, BASE_URL + '/lookup-user');
+            assert.equal(request.type, 'get');
+            assert.include(request.headers.Authorization, 'token');
+          });
+      });
+
+      it('returns a `null` if no token returned for the user', function () {
+        sinon.stub(xhrMock, 'ajax', function () {
+          return p({});
+        });
+
+        return client.getPreferencesUrl('token')
+          .then(function (url) {
+            assert.isNull(url);
+          });
+      });
+    });
+
+    describe('optIn', function () {
+      it('opts the user in', function () {
+        sinon.spy(xhrMock, 'ajax');
+
+        return client.optIn('token')
+          .then(function () {
+            var request = xhrMock.ajax.args[0][0];
+            assert.equal(request.url, BASE_URL + '/subscribe');
+            assert.equal(request.type, 'post');
+            assert.include(request.headers.Authorization, 'token');
+          });
+      });
+    });
+
+    describe('optOut', function () {
+      it('opts the user out', function () {
+        sinon.spy(xhrMock, 'ajax');
+
+        return client.optOut('token')
+          .then(function () {
+            var request = xhrMock.ajax.args[0][0];
+            assert.equal(request.url, BASE_URL + '/unsubscribe');
+            assert.equal(request.type, 'post');
```

Comment:
Is it worth checking (and can we easily check?) the expected request body as part of this test?